litmus-portal/backend/graphql-server/manifests/subscriber.yml

```diff
@@ -24,3 +24,11 @@ spec:
               value: #{CID}
             - name: KEY
               value: #{KEY}
+---
+apiVersion: v1
```

Comment:
If the manifest is reapplied the configmap data will also be re-configured?

Reply:
Yes, it will re-configured with this config

Reply:
So we cannot have the user reapply this, we need to create this config map only after the subscriber comes up? Because if they unknowingly reapply the whole thing is reconfigured and identity of the cluster changes? We should deploy the subscriber and do the pre-check inside it and if it's a new install we create the configmap through the subscriber directly.

Reply:
What if the user reapply subscriber, it will also reconfigured the confimap config?

Reply:
Future work- if user reapplies this subscriber manifest, the subscriber will call the cluster confirmed mutation and if cluster_confimed is true in MongoDB, it will send the new key, which then it will update it in config map.

Reply:
Yes it would reconfigure the subscriber but the configmap will remain untouched, so when the subscriber restarts it will get the old details and will be able to verify that the cluster is already registered so it won't re-register the cluster, it'll skip that step and use the old identity.

Reply:
For now, it's okay to have it here. After that, we need to modify cluster confirm mutation

Reply:
The behavior of cluster-confirm is the same as you have stated, if we send a cluster-confirm request again it will update the key. That is not a problem, but as the subscriber will be responsible for installing all the other components if it is restarted and has no context of whether there is already an installation for the same, it will reinstall everything from scratch that might cause some downtime considering a workflow was scheduled right at that moment.

Reply:
That would also mean any previous workflow/experiment schedules will be re-configured or removed too.